backend/src/org/commcare/util/FormDataUtil.java

```diff
@@ -48,12 +48,16 @@ public static String getTitleFromSession(UserSandbox userSandbox,
                 // the 1st computed datum... another good heuristic would be
                 // session.getPoppedStep().getId().startsWith("case_id")
                 datumValue = session.getPoppedStep().getValue();
-            } else if (datum instanceof EntityDatum && ((EntityDatum)datum).getLongDetail() != null) {
+            } else if (datum instanceof EntityDatum) {
                 String tmpDatumValue = session.getPoppedStep().getValue();
                 if (tmpDatumValue != null) {
                     datumValue = tmpDatumValue;
                 }
-                return new Pair<>((EntityDatum)datum, datumValue);
+                if (((EntityDatum)datum).getLongDetail() == null) {
+                    return new Pair<>(null, datumValue);
```

Comment:
Gotcha. I would advocate for adding some sort of comment here to indicate that, and maybe renaming the method a bit as well. With the method just called `getDatumAndCaseId`, it appears counterintuitive that you wouldn't return the datum just because it's missing a long detail